torch/csrc/autograd/python_variable.cpp

```diff
@@ -333,7 +333,13 @@ class PyInterpreterHolder {
  public:
   PyInterpreterHolder()
       : impl_(new c10::impl::PyInterpreter(
-            ConcretePyInterpreterVTable::instance())) {}
+            ConcretePyInterpreterVTable::instance())) {
+#ifndef USE_DEPLOY
+    // If this is not a torchdeploy interpreter, register this as the global
+    // PyInterpreter
+    c10::impl::PyInterpreterTLS::set_global_state(impl_);
```

Comment:
as long as we can call set_state from deploy and override this setting should be fine and don't need the USE_DEPLOY